DependencyInjection/Configuration.php

```diff
@@ -368,34 +370,61 @@ private function addSharedPluginNodes(ArrayNodeDefinition $pluginNode, $disableA
                         ->cannotBeEmpty()
                     ->end()
                     ->arrayNode('config')
-                        ->addDefaultsIfNotSet()
                         ->validate()
-                        ->ifTrue(function ($config) {
-                            // Cannot set both respect_cache_headers and respect_response_cache_directives
-                            return isset($config['respect_cache_headers'], $config['respect_response_cache_directives']);
-                        })
-                        ->thenInvalid('You can\'t provide config option "respect_cache_headers" and "respect_response_cache_directives" simultaniously. Use "respect_response_cache_directives" instead.')
+                            ->ifTrue(function ($config) {
+                                // Cannot set both respect_cache_headers and respect_response_cache_directives
+                                return isset($config['respect_cache_headers'], $config['respect_response_cache_directives']);
+                            })
+                            ->thenInvalid('You can\'t provide config option "respect_cache_headers" and "respect_response_cache_directives" simultaniously. Use "respect_response_cache_directives" instead.')
                         ->end()
                         ->children()
-                            ->scalarNode('default_ttl')->defaultValue(0)->end()
+                            ->enumNode('cache_key_generator')
+                                ->info('A class implementing '.CacheKeyGenerator::class.' to generate a PSR-6 cache key')
+                                ->values([null, CacheKeyGenerator::class])
+                            ->end()
+                            ->scalarNode('cache_lifetime')
+                                ->info('The minimum time we should store a cache item')
+                            ->end()
+                            ->scalarNode('default_ttl')
```

Comment:
probably because support for Symfony 2.3 was needed in the past, which did not have the integer node yet.